src/librustc_mir/borrow_check/nll/type_check/mod.rs

```diff
@@ -776,6 +776,7 @@ impl<'a, 'gcx, 'tcx> TypeChecker<'a, 'gcx, 'tcx> {
             | StatementKind::InlineAsm { .. }
             | StatementKind::EndRegion(_)
             | StatementKind::Validate(..)
+            | StatementKind::UserAssertTy(..)
```

Comment:
I think we want to do a check here -- something pretty similar to what `Assign` does. i.e., get the type of the `place` whose type is being asserted and invoke `self.sub_types()` (or actually `eq_type` perhaps).

Reply:
I intended to add a check here in a future commit, I initially included the variant here just so I could see if being included in the MIR without it doing anything.

Reply:
seems fine

Reply:
Added something that uses `eq_type`, not sure it's doing 100% of what it needs to though.